### PR TITLE
feat: Add usePSP22Balances

### LINKED ISSUE #25: Add `usePSP22Balances`
Add a custom React hook to get PSP-22 token balances.

### PR DESCRIPTION
**Pull request by ical10**
Closes #25 

## Included in this PR

- [x] `usePSP22Balances`, a custom React hook to fetch PSP-22 tokens balance
- [x] `psp22Asset.json`, a static list of supported PSP-22 tokens (currently supporting INW and PANX) 
- [x] Added `watch` balance capability: balance will update once a transfer is confirmed

### CODE REVIEW

File: src/helpers/getPSP22Balances.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  return () => {
+    clearInterval(intervalId)
+  }
+}
+
+/**
+ * Helper to parse the fetched PSP22 token balance data.
+ */
+
+export const parsePSP22Balance = (
```
Comment (wottpal) [inline src/helpers/getPSP22Balances.ts:171]: This function is a bit odd, it takes the same data (`PSP22BalanceData`) as it returns (`PSP22BalanceData`). Just one w/o balanceFormatted and one with which is is only implicitly noticeable...

Suggestion: Replace it `formatPSP22Balance` that takes `PSP22BalanceData` (and optionally `formatterOptions`) that **only** returns the `balanceFormatted` string which you then add to the `PSP22BalanceData` object outside of the function.

File: src/types.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export enum AssetType {
+  PSP22 = 'PSP22',
+  PSP34 = 'PSP34',
+  UNKNOWN = 'UNKNOWN',
```
Comment (wottpal) [inline src/types.ts:4]: Why do we need this?

File: src/psp22_abi.json
Comment (wottpal) [inline src/psp22_abi.json:1]: We should decrease the file size of this. Please:

- [ ] Remove all the `docs` attributes
- [ ] Use the minification script we're using in our contracts repository to remove whitespace: https://github.com/azero-id/contracts/blob/develop/minify-metadata.sh

**Review by wottpal — CHANGES_REQUESTED** [review]

File: src/helpers/getPSP22Balances.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+  tokenDecimals: number
+  tokenSymbol: string
+  iconPath: string
+  balance?: BN
+  balanceFormatted?: string
+}
+/**
+ * Default refresh interval for the PSP-22 token balances.
+ */
+
```
[the remark this answers is not in the document]
Comment (ical10) [reply src/helpers/getPSP22Balances.ts]: Removed in 7025175

File: src/helpers/getPSP22Balances.ts [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+export const PSP22_TOKEN_BALANCE_SUBSCRIPTION_INTERVAL = 60000
+
+/**
+ * Returns the PSP-22 token balances of the given `address`.
+ */
+
```
[the remark this answers is not in the document]
Comment (ical10) [reply src/helpers/getPSP22Balances.ts]: 7025175

File: src/helpers/getAbi.ts
[the remark this answers is not in the document]
Comment (ical10) [reply src/helpers/getAbi.ts:1]: - [x] Changed to ESM imports in 7025175
- [x] Added a small comment for reference to Subwallet's JSON (https://github.com/scio-labs/use-inkathon/pull/26/commits/7025175cb1d11d4ebfb0156c7e754b35f5e8369d#diff-9bb89c713967fadb1f51f27cfa4b5202055f2c775176e272856acbf2e8da6db1R2)

File: src/helpers/getPSP22Balances.ts (hunk 1 above)
Comment (ical10) [reply src/helpers/getPSP22Balances.ts:172]: Replaced in 7025175

File: src/helpers/getPSP22Balances.ts [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+          const psp22BalanceData = parsePSP22Balance(data, formatterOptions)
+          return psp22BalanceData
+        }),
+      ),
+    )
+
+  fetchTokenBalances()
+
+  // Create intervalId which can be used to unsubscribe
+  const intervalId = setInterval(fetchTokenBalances, PSP22_TOKEN_BALANCE_SUBSCRIPTION_INTERVAL)
```
[the remark this answers is not in the document]
Comment (ical10) [reply src/helpers/getPSP22Balances.ts:162]: I'll do more research next week to find a cleaner way. Got it, won't activate it for token widgets.

File: src/types.ts (hunk 2 above)
Comment (ical10) [reply src/types.ts]: Removed in 7025175

File: src/helpers/getPSP22Balances.ts [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+  return () => {
+    clearInterval(intervalId)
+  }
+}
+
+/**
+ * Helper to parse the fetched PSP22 token balance data.
+ */
+
```
[the remark this answers is not in the document]
Comment (ical10) [reply src/helpers/getPSP22Balances.ts]: Removed in 7025175

File: src/psp22_abi.json
Comment (ical10) [reply src/psp22_abi.json:1]: File size was decreased per your suggestions (from 20ish to 8 KB). See 064eac4.

**Review by wottpal — CHANGES_REQUESTED** [review]

File: src/types.ts [hunk 7]
```diff
@@ -1,16 +1,9 @@
-export enum AssetType {
-  PSP22 = 'PSP22',
-  PSP34 = 'PSP34',
-  UNKNOWN = 'UNKNOWN',
-}
```
[the remark this answers is not in the document]
Comment (ical10) [reply src/types.ts]: added back in f9369b8 with removing UNKNOWN type.

File: src/helpers/getAbi.ts [hunk 8]
```diff
@@ -1,2 +1,2 @@
-// eslint-disable-next-line @typescript-eslint/no-var-requires,@typescript-eslint/no-unsafe-assignment
-export const _PSP22_ABI: Record<string, any> = require('../psp22_abi.json')
+import PSP22_ABI from '../psp22_abi.json'
+export const _PSP22_ABI: Record<string, any> = PSP22_ABI
```
[the remark this answers is not in the document]
Comment (ical10) [reply src/helpers/getAbi.ts]: Changed the casing to not have underscore (also identical to casings in `getPSP22Asset.ts`)

**Pull request merged**